src/rpc/SaslClient.cpp

```diff
@@ -39,9 +39,11 @@ namespace Internal {
 
 SaslClient::SaslClient(const RpcSaslProto_SaslAuth & auth, const Token & token,
                        const std::string & principal) :
-    complete(false) {
+    complete(false), theAuth(auth), theToken(token), thePrincipal(principal) {
     int rc;
     ctx = NULL;
+    privacy = false;
+    integrity = false;
```

Comment:
It leads to double initialization. At first the variable is created and initalized, than in constructor body you assign a new value to it. It's not so bad for boolean values but not very nice for complex objects.